client/my-sites/checkout/concierge-session-nudge/index.jsx

```diff
@@ -63,13 +62,63 @@ export class ConciergeSessionNudge extends React.Component {
 	footer() {
 		return (
 			<footer className="concierge-session-nudge__footer">
-				<Button className="concierge-session-nudge__decline-offer-button">No, thank you.</Button>
-				<Button primary className="concierge-session-nudge__accept-offer-button">
+				<Button
+					className="concierge-session-nudge__decline-offer-button"
+					onClick={ this.handleClickDecline }
+				>
+					No, thank you.
+				</Button>
+				<Button
+					primary
+					className="concierge-session-nudge__accept-offer-button"
+					onClick={ this.handleClickAccept }
+				>
 					Yes, please.
 				</Button>
 			</footer>
 		);
 	}
+
+	handleClickDecline = () => {
+		const { siteSlug, receiptId, isEligibleForChecklist, trackUpsellButtonClick } = this.props;
+
+		trackUpsellButtonClick( 'decline' );
+
+		page(
+			isEligibleForChecklist
+				? `/checklist/${ siteSlug }`
+				: `/checkout/thank-you/${ siteSlug }/${ receiptId }`
+		);
+	};
+
+	handleClickAccept = () => {
+		const { siteSlug, trackUpsellButtonClick } = this.props;
+
+		trackUpsellButtonClick( 'accept' );
+
+		addItem( cartItems.conciergeSessionItem() );
+
+		page( `/checkout/${ siteSlug }` );
+	};
 }
 
-export default connect()( localize( ConciergeSessionNudge ) );
+const trackUpsellButtonClick = buttonAction => {
+	// Track calypso_concierge_session_upsell_decline_button_click and calypso_concierge_session_upsell_accept_button_click events
+	return recordTracksEvent( `calypso_concierge_session_upsell_${ buttonAction }_button_click`, {
+		section: 'checkout',
+	} );
+};
+
+export default connect(
+	( state, props ) => {
+		const { selectedSiteId } = props;
+
+		return {
+			siteSlug: getSiteSlug( state, selectedSiteId ),
+			isEligibleForChecklist: isEligibleForDotcomChecklist( state, selectedSiteId ),
+		};
+	},
+	{
+		trackUpsellButtonClick,
+	}
+)( localize( ConciergeSessionNudge ) );
```

Comment:
I don't think `localize` wrapper is necessary for now since the component translates no strings.